### PR TITLE
Adds global theme string support

### PR DESCRIPTION
**Pull request by jbrundage**
Using a global variable, window.__hpcc_theme, to declare the theme string to be applied to any/all widgets.

Fixes GH-685

@GordonSmith Please review

Is this a preferable structure for the theme serial? If not, what do you suggest?

Signed-off-by: Jay Brundage <jaman.brundage@lexisnexis.com>

### CODE REVIEW

**Comment by jbrundage** [comment]
@GordonSmith This is ready to be inspected again.

I've added the *Theme dropdown* along with a few themes to dermatology.html.

I've also added the Theme Text checkbox.

https://rawgit.com/jbrundage/Visualization/Theme_Global/demos/dermatology.html?src/layout/Grid

File: src/other/Persist.js [hunk 1]
```diff
@@ -31,7 +31,144 @@
...
+                                    var paramIsIncluded = false;
+                                    var classParts = widget._class.trim().split(" ");
+                                    classParts.forEach(function(c) { 
+                                        if(!paramIsIncluded){
+                                            if(propObj[c] === undefined){
+                                                propObj[c] = {};
+                                            } 
+                                            if (propObj[c][item.id] === undefined) { 
+                                                propObj[c][item.id] = widget[item.id](); 
+                                                paramIsIncluded = true;
```
Comment (GordonSmith) [inline src/other/Persist.js]: You can break out of the loop at this point?

File: src/other/Persist.js [hunk 2]
```diff
@@ -31,7 +31,144 @@
...
+                                            } 
+                                            else if (propObj[c][item.id] === widget[item.id]()) {
+                                                paramIsIncluded = true;
```
Comment (GordonSmith) [inline src/other/Persist.js]: You can break out of the loop at this point?

File: src/other/Persist.js [hunk 3]
```diff
@@ -31,7 +31,144 @@
...
+                                }
+                                break;
+                        }
+                    }
+                });
+                function _isFilterMatch(str,arr){
+                    var ret = false;
+                    arr.forEach(function(substr){
+                        if(str.indexOf(substr) !== -1){
+                            ret = true;
```
Comment (GordonSmith) [inline src/other/Persist.js]: You should break out of the forEach at this point.

File: src/other/Persist.js (hunk 1 above)
Comment (jbrundage) [inline src/other/Persist.js]: As it turns out... there's no built-in break for Array.forEach. I could turn it into a **for(var i in classParts)** though.

Comment (GordonSmith) [inline src/other/Persist.js]: You should use Array.some I think (or for)

Comment (jbrundage) [inline src/other/Persist.js]: I've switched them to **for** just now. Array.some sounds interesting, I havent used it before.

**Comment by GordonSmith** [comment]
@jbrundage With those "break" changes your theme strings should be more concise, can you update the ones you use in the dermatology page?

**Comment by jbrundage** [comment]
@GordonSmith They should produce the same strings. This boolean **paramIsIncluded** was being used to break without actually breaking. I should remove it now that it's actually using break.

**Comment by jbrundage** [comment]
@GordonSmith I believe this is ready to be pulled

**Comment by GordonSmith** [comment]
Looks good - can you remove the dermotology_1.html file?

**Comment by jbrundage** [comment]
@GordonSmith It's removed

**Pull request merged**